src/main/java/de/danielbechler/diff/selector/CollectionItemElementSelector.java

```diff
@@ -71,6 +94,6 @@ public boolean equals(final Object o)
 	@Override
 	public int hashCode()
 	{
-		return item != null ? item.hashCode() : 0;
+		return item != null ? identityStrategy.hashCode(item) : 0;
```

Comment:
Yep, and the only reason this is a problem is because of the way the ElementSelector is looked up.  It's a very low hanging fruit to remove this source of confusion, because it only takes changing one line of code and a small (in most cases probably unnoticeable) performance impact when traversing the graph of changed nodes. The library doesn't need the `hashCode` and the user is only interested in changing the behavior of equals.